pkg/controller/deployment/deploment_helper_test.go

```diff
@@ -0,0 +1,398 @@
+					corev1.ResourceMemory: k8sresource.MustParse("128Mi"),
+				},
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("10m"),
+					corev1.ResourceMemory: k8sresource.MustParse("32Mi"),
+				},
+			},
+			expected: corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("500m"),
+					corev1.ResourceMemory: k8sresource.MustParse("128Mi"),
+				},
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("10m"),
+					corev1.ResourceMemory: k8sresource.MustParse("32Mi"),
+				},
+			},
+		},
+		{
+			name: "override resources only replaces source resource limits, doesn't replace source requests",
+			sourceResources: corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("400m"),
+					corev1.ResourceMemory: k8sresource.MustParse("64Mi"),
+				},
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("10m"),
+					corev1.ResourceMemory: k8sresource.MustParse("32Mi"),
+				},
+			},
+			overrideResources: v1alpha1.CertManagerResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("500m"),
+					corev1.ResourceMemory: k8sresource.MustParse("128Mi"),
+				},
+			},
+			expected: corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("500m"),
+					corev1.ResourceMemory: k8sresource.MustParse("128Mi"),
+				},
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("10m"),
+					corev1.ResourceMemory: k8sresource.MustParse("32Mi"),
+				},
+			},
+		},
+		{
+			name: "override resources doesn't replace source resource limits, replaces source requests",
+			sourceResources: corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("400m"),
+					corev1.ResourceMemory: k8sresource.MustParse("64Mi"),
+				},
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("10m"),
+					corev1.ResourceMemory: k8sresource.MustParse("32Mi"),
+				},
+			},
+			overrideResources: v1alpha1.CertManagerResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("40m"),
+					corev1.ResourceMemory: k8sresource.MustParse("64Mi"),
+				},
+			},
+			expected: corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("400m"),
+					corev1.ResourceMemory: k8sresource.MustParse("64Mi"),
+				},
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("40m"),
+					corev1.ResourceMemory: k8sresource.MustParse("64Mi"),
+				},
+			},
+		},
+		{
+			name: "override resources limits and requests merges with source resource limits and requests respectively",
+			sourceResources: corev1.ResourceRequirements{
```

Comment:
Can you add test cases for testing limits and request as nil explicitly?

Similar to:
```
{
			name: "override resources limits and requests merges with source resource limits and requests respectively",
			sourceResources: corev1.ResourceRequirements{
				Limits: nil,
				Requests: corev1.ResourceList{
					corev1.ResourceCPU: k8sresource.MustParse("10m"),
				},
			},
			overrideResources: v1alpha1.CertManagerResourceRequirements{
				Limits: corev1.ResourceList{
					corev1.ResourceCPU: k8sresource.MustParse("400m"),
				},
				Requests: corev1.ResourceList{
					corev1.ResourceMemory: k8sresource.MustParse("64Mi"),
				},
			},
			expected: corev1.ResourceRequirements{
				Limits: corev1.ResourceList{
					corev1.ResourceCPU: k8sresource.MustParse("400m"),
				},
				Requests: corev1.ResourceList{
					corev1.ResourceCPU:    k8sresource.MustParse("10m"),
					corev1.ResourceMemory: k8sresource.MustParse("64Mi"),
				},
			},
		}
```